emmet-builders/emmet/builders/mobility_screening/migration_graph.py

```diff
@@ -0,0 +1,118 @@
+from emmet.core.vasp.task_valid import TaskDocument
+from pymatgen.analysis.diffusion.neb.full_path_mapper import MigrationGraph
+from pymatgen.entries.computed_entries import ComputedEntry, ComputedStructureEntry
+from maggma.stores import MongoStore
+from maggma.builders import MapBuilder
+
+
+class MigrationGraphBuilder(MapBuilder):
+    def __init__(
+        self,
+        electrodes: MongoStore,
+        tasks: MongoStore,
+        migration_graphs: MongoStore,
+        ltol: float = 0.4,
+        stol: float = 0.6,
+        angle_tol: float = 15,
+        **kwargs
+    ):
+        """
+        Attach corresponding migration graphs to each electrode document by getting its computed structure
+            entry from an auxilarry task store.
+        Args:
+        electrodes (Store): Store of electrode documents that contains task_ids for its materials
+        tasks (Store): Store of task documents that contain the structure entry, to be paired with
+            each electrode docs based on material ids
+        migration_graphs (Store): Target store for electrode docs with attached migration graph
+        query (dict): dictionary to limit materials to be analyzed ---
+            only applied to the materials when we need to group structures
+            the phase diagram is still constructed with the entire set
+        ltol: fractional length tolerance for StructureMatcher
+        stol: site tolerance for StructureMatcher
+        angle_tol: angle tolerance for StructureMatcher
+        """
+        self.electrodes = electrodes
+        self.tasks = tasks
+        self.ltol = ltol
+        self.stol = stol
+        self.angle_tol = angle_tol
+        self.migration_graphs = migration_graphs
+        self.wi_entries = {}
+        from pymatgen.ext.matproj import MPRester
```

Comment:
The import statements for MPRester and MaterialsProject2020Compatibility should go at the top of the file. We typically do not put import statements within the builder, just at the start of the file.